### ISSUE TITLE
Accept presentation from another application frame.

### ISSUE THREAD

**Issue by admich** [robert-strandh/McCLIM#294]
Is it possible to accept presentation from another application frame?

For example :

1.  start a listener
2.  show directory in the listener
3.  open Climacs
4.  C-x C-f in climacs to open a file
5.  mouse click on a pathname presentation in the listener

I try but it doesn't work.

I play a little bit with [opengenera](https://github.com/ynniv/opengenera) and in that system it was/is possible.

I think this is an important feature to increase the interoperability between clim applications.

**Comment by dkochmanski** [comment]
It looks like a very appealing feature. How Genera handles situation, where two application frames have commands waiting for the argument of the same type?

Also, I'm not 100% sure if climacs accepts presentation of the same type listener presents its files. We'd have to check that. I'd test with some dull type, like integer or form, which is sure to be common for both application frames.

**Comment by admich** [comment]
As I understand in Genera at each moment there is an "active" application. All the visible (but the windows can be scrolled with the scroll bars) presentations accepted from the active application can be clicked and the input go to the active application. For example if i have two Listener and Listener1 is active and i click on a form presentation in Listener2, the form  is executed in Listener1. If Listener1 wait for a pathname (for example with Show Directory command), then the form presentations in Listener2 are not active, only the pathnames are active. If i click on a pathname in Listener2 the input goes to Listener1 if i click in another region the active application change and Listener2 becomes active.

**Comment by dkochmanski** [comment]
This is just a sketch of an idea. We plan to introduce a term of a
"focused sheet", that is a sheet, which receives keyboard events. This
concept may be extended to consider a focused sheet as a sheet which
event-queue is used in ~dispatch-event~ (event sheet does not change,
only the queue). Usually all sheets in a frame use the same
event-queue, so the event will be read in the input context of the
focused sheet, hence it will be possible to return a
presentation. Otherwise we may call ~handle-event event-sheet sheet~
(or, alternatively, queue the event in the target sheet queue, to
avoid asynchronous access - mind that schedule-event may schedule an
event even if we do not dispatch to said queue).

**Comment by admich** [comment]
I agree with this sketch, this is similar to some experiment that I made several month ago: https://github.com/admich/omni-accept 
One difficult part is how to control the change of the "focused sheet". It can be changed from clim but also with some window manager shurtcut key or with mouse.

**Comment by dkochmanski** [comment]
there is set-stream-input-focus (and `with-input-focus`) for the programmatic part. As of window reactions, please see:

- https://github.com/McCLIM/McCLIM/wiki/Input-focus-proposal
- https://github.com/McCLIM/McCLIM/tree/improve-click-to-focus
